### PR TITLE
Configure web-password

### PR DESCRIPTION
**Pull request by konsumer**
This will configure the web-password, if the environment variable `WEBPASSWD` is set.

### CODE REVIEW

File: init/91_webpasswd.sh [hunk 1]
```diff
@@ -0,0 +1,3 @@
+if [ -z "$WEBPASSWD" ];then
```
Comment (cpoppema) [inline init/91_webpasswd.sh]: I am no bash hero, but according to the test's manpage, -z returns TRUE when the string being tested is empty, so shouldn't there be an exclamation mark ?

```
$ if [ -z "" ]; then echo "set password"; else echo "do nothing"; fi;
set password
```
```
$ if [ -z "p4ssw0rd" ]; then echo "set password"; else echo "do nothing"; fi;
do nothing
```

Comment (konsumer) [reply init/91_webpasswd.sh]: ah, yes, sorry

**Comment by cpoppema** [comment]
Thanks for the suggestion and both pull requests! I have only one reservation which we can hopefully resolve: the env-variable is set when the container is created. This means, if I'm not wrong, that when you change the password using the `docker exec` command, your password will be reset to what it was when creating the container when you restart that container. Is that intentional ?

**Comment by konsumer** [comment]
Yeah, maybe the `if` statement should be in the `RUN`...

**Comment by cpoppema** [comment]
Sorry to get back to you so late, but moving it from `91_webpasswd.sh` to `run` doesn't actually change the behavior. Both are executed every time upon `docker start`, overwriting any password you might've set in a previous `docker exec`.

A possible workaround is to create a "lock"-file when executing `91_webpasswd.sh` and setting the password once, a simple `touch password.lock` is enough. If you skip setting a password when that lock file exists, only the first `docker start` will set it from the environment variable. Just don't put it in /tmp (this is cleaned automatically once in a while or on system start), but I don't see any harm to put it in `/config/` (it is only ever used when somebody sets the environment variable at all after all) - and this probably also survives recreating containers which is I guess a good thing.

**Comment by konsumer** [comment]
Hmm, I thought things in `run` were only run when the local container is built. I'll take another look.

**Comment by cpoppema** [comment]
Hi @konsumer, I'm closing this pull request after merging https://github.com/cpoppema/docker-flexget/commit/5f5811ceba061ed616a4e42d120d54b4b2c26901.

**Comment by konsumer** [comment]
Thanks!

**Pull request closed without merging**